bot.py

```diff
@@ -243,97 +256,27 @@ def addsocials(message):  # the message is the social media message
-            break
-
-    for url in socials:
-        if "instagram" in url:
-            instagram = url
-            socials.remove(url)
-            break
-
-    for url in socials:
-        if "steam" in url:
-            steam = url
-            socials.remove(url)
-            break
+    author_map = {
+        "authorNick": author.display_name,
+        "authorsAvatarUrl": str(author.avatar.url),
+    }
 
-    for url in socials:
-        if "twitter" in url:
-            twitter = url
-            socials.remove(url)
-            break
+    for key in known_socials:
+        for url in socials:
+            if key in url:
+                author_map[key] = url
+                socials.remove(url)
+                break
 
+    author_map["othersocials"] = socials
     authorid_str = str(author.id)
 
     if authorsdb.search(Query().authorid == authorid_str) != []:
-        authorsdb.update(
-            {
-                "authorNick": author.display_name,
-                "authorsAvatarUrl": str(author.avatar.url),
-                "othersocials": socials,
-            },
-            Query().authorid == authorid_str,
-        )  # avatar and nick update
-        if flickr != "":
-            authorsdb.update(
-                {
-                    "authorNick": author.display_name,
-                    "authorsAvatarUrl": str(author.avatar.url),
-                    "flickr": flickr,
-                },
-                Query().authorid == authorid_str,
-            )
-        if instagram != "":
-            authorsdb.update(
-                {
-                    "authorNick": author.display_name,
-                    "authorsAvatarUrl": str(author.avatar.url),
-                    "instagram": instagram,
-                },
-                Query().authorid == authorid_str,
-            )
-        if twitter != "":
-            authorsdb.update(
-                {
-                    "authorNick": author.display_name,
-                    "authorsAvatarUrl": str(author.avatar.url),
-                    "twitter": twitter,
-                },
-                Query().authorid == authorid_str,
-            )
-        if steam != "":
-            authorsdb.update(
-                {
-                    "authorNick": author.display_name,
-                    "authorsAvatarUrl": str(author.avatar.url),
-                    "steam": steam,
-                },
-                Query().authorid == authorid_str,
-            )
+        authorsdb.update(author_map, Query().authorid == authorid_str)  # avatar and nick update
     else:
-        authorsdb.insert(
-            {
-                "authorNick": author.display_name,
-                "authorid": authorid_str,
-                "authorsAvatarUrl": str(author.avatar.url),
-                "flickr": flickr,
-                "twitter": twitter,
-                "instagram": instagram,
-                "steam": steam,
-                "othersocials": socials,

```

Comment:
Thinking about this a bit more after working on updating the front end — why bother maintaining this distinct set of sets in addition to the `othersocials` list? The front end uses the URL domain to determine which icon to display anyway, this could be simplified by just having a `socials` key that's just an array of strings.

Not sure what kind of a pain it would be to migrate existing data though, just a thought.